pymilo/transporters/neighbors_tree_transporter.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+"""PyMilo Tree(from sklearn.tree._tree) object transporter."""
+from sklearn.neighbors._kd_tree import KDTree
+from sklearn.neighbors._ball_tree import BallTree
+
+from .transporter import AbstractTransporter
+
+type_to_tree = {
+    "KDTree": KDTree,
+    "BallTree": BallTree
+}
+
+
+class NeighborsTreeTransporter(AbstractTransporter):
+    """Customized PyMilo Transporter developed to handle (pyi,pyx) NeighborsTreeTransporter object."""
+
+    def serialize(self, data, key, model_type):
+        """
+        Serialize the special _tree field of the Neighbors model.
+
+        :param data: the internal data dictionary of the given model
+        :type data: dict
+        :param key: the special key of the data param, which we're going to serialize its value(data[key])
+        :type key: object
+        :param model_type: the model type of the ML model
+        :type model_type: str
+        :return: pymilo serialized output of data[key]
+        """
+        if isinstance(data[key], KDTree):
+            data[key] = {
+                'pymilo-bypass': True,
+                'pymilo-tree-type': "KDTree",
+            }
+        elif isinstance(data[key], BallTree):
+            data[key] = {
+                'pymilo-bypass': True,
+                'pymilo-tree-type': "BallTree",
+            }
+        return data[key]
+
+    def deserialize(self, data, key, model_type):
+        """
+        Deserialize the special _tree field of the Neighbors model.
+
+        The associated tree_ field of the pymilo serialized model, is extracted through
+        it's previously serialized parameters.
+        deserialize the data[key] of the given model which type is model_type.
+        basically in order to fully deserialize a model, we should traverse over all the keys of its serialized data dictionary and
+        pass it through the chain of associated transporters to get fully deserialized.
```

Comment:
That's too much for doc-string. I think we you can simplify it into 1/2 sentence(s).

Reply:
Done.
I will also apply a deep refactoring in docstrings in a separate enhancing pr.
@sadrasabouri